docs/content/preview/architecture/transactions/concurrency-control.md

```diff
@@ -1223,6 +1221,36 @@ commit;
 </tbody>
 </table>
 
+### Versioning and upgrades
+
+When turning `enable_wait_queues` on or off, or during a rolling restart, where during an update the flag could be on on nodes with a more recent version, if some nodes have wait-on-conflict behavior enabled and some don’t, you will experience mixed (but still correct) behavior.
+
+A mix of both fail-on-conflict and wait-on-conflict traffic results in the following additional YSQL-specific semantics:
+
+- If a transaction using fail-on-conflict encounters transactions that have conflicting writes -
+    - If there is even a single conflicting transaction that uses wait-on-conflict, the transaction aborts.
+    - Else, YugabyteDB will abort either the current request or encountered conflicting transactions based on transaction priority
```

Comment:
```suggestion
    - Otherwise, YugabyteDB uses the regular [fail-on-conflict resolution](#fail-on-conflict), which is to abort the lower priority transaction.
```